### PR TITLE
interfaces: adjust steam to allow all apparmor permissions

### PR DESCRIPTION
**Pull request by zyga**
This is implemented by @jrjohansen but I will garden tests and ensure it can land.

Technically the feature works as is today but can benefit from AppArmor 4.0.2 (unreleased). We will most likely want to add https://github.com/snapcore/snapd/pull/14167 to ensure that we can use apparmor from the host to the full extent and to create a new PR updating bundled apparmor (inside snapd snap) once a release is available upstream.

### CODE REVIEW

**Comment by zyga** [comment]
I've updated the 2nd patch and shared it with @jrjohansen internally but I cannot push it to the PR.

File: sandbox/apparmor/apparmor.go [hunk 1]
```diff
@@ -673,6 +673,11 @@ func probeParserFeatures() ([]string, error) {
 			probe:   "mqueue,",
 			minVer:  "4.0.1",
 		},
+		{
+			feature: "all",
```
Comment (alexmurray) [inline sandbox/apparmor/apparmor.go:677]: `all` seems a bit terse - how about `allow-all`?

**Review by alexmurray — APPROVED** [review]
LGTM - one minor nitpick on the feature name.

File: sandbox/apparmor/apparmor.go (hunk 1 above)
Comment (zyga) [reply sandbox/apparmor/apparmor.go:677]: Yeah. Good point.

**Comment by zyga** [comment]
I'm closing this in favour of https://github.com/snapcore/snapd/pull/14216 since I cannot push any changes here.

File: interfaces/builtin/steam_support.go [hunk 2]
```diff
@@ -395,6 +247,25 @@ type steamSupportInterface struct {
 	commonInterface
 }
 
+func (iface *dockerSupportInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	// if apparmor supports allow all us it. This allows not updating
```
Comment (jslarraz) [inline interfaces/builtin/steam_support.go:251]: nit: us it -> use it

**Pull request closed without merging**